realm/realm-library/src/main/java/io/realm/RealmObject.java

```diff
@@ -61,10 +52,8 @@
  * The types <code>short</code>, <code>int</code>, and <code>long</code> are mapped to <code>long</code> when storing
  * within a Realm.
  * <p>
- * Getter and setter names must have the name {@code getXXX} or {@code setXXX} if the field name is {@code XXX}. Getters
- * for fields of type boolean can be called {@code isXXX} as well. Fields with a m-prefix must have getters and setters
- * named setmXXX and getmXXX which is the default behavior when Android Studio automatically generates the getters and
- * setters.
+ * You can use any name for getters and setters as long as they are valid method name in Java language, or
```

Comment:
I would move away from talking about getters and setters since that is really just a subset of what you can do. Perhaps

"The only restriction a RealmObject has is that fields are not allowed to be `final`, `transient'` or `volatile`. Any method as well as public fields are allowed."